Plugins/Object/Object.cpp

```diff
@@ -346,4 +347,18 @@ ArgumentStack Object::CheckFit(ArgumentStack&& args)
     Services::Events::InsertArgument(stack, retVal);
     return stack;
 }
+
+ArgumentStack Object::GetDamageImmunity(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+    int32_t retVal = -1;
+
+    if (auto *pObject = object(args))
+    {
+        const uint16_t damageFlags = Services::Events::ExtractArgument<int32_t>(args);
```

Comment:
Minor thing, but you can use `auto` instead of `uint16_t` and let the compiler figure out what's best.

Since `GetDamageImmunityByFlags()` takes an unsigned int, I'd also throw in a `ASSERT_OR_THROW(damageFlags >= 0);`